woocommerce/wc-admin
# Docs Project: Add markdown files, docsify to render site

This is the final PR, building on #334 & #336. This PR includes setup for a docsify site and the generated markdown files (which is why the diff is so large, and between this & copying the inline comments, why I created 3 PRs).

The goal of this set of PRs is to have a documentation source that third-party devs (and us) can look at for referencing the components exported on `wp.components`. We haven't been great at keeping READMEs up-to-date, but we're pretty good at propTypes. `react-docgen` can extract info from PropTypes & inline documentation, which is what we're doing in #336. Once we have that documentation info, we build it into markdown files that live in `/docs/components`.

Now we can use [`docsify`](https://docsify.js.org/) to render the files in `/docs` in a webpage. As long as we keep the inline docs up to date, which will be easier to check in PRs, the documentation will be correct. 🎉

We can even customize the docsify page with themes & plugins, so I've added the text search & customized the theme to be woo-flavored:

![docs](https://user-images.githubusercontent.com/541093/44885485-18badb80-ac8f-11e8-813e-3be0f3a543e6.png)

**To test**

- Run the docsify server locally: `npm install`, then `npx docsify serve ./docs`
- Make sure the documentation matches reality
- Change some documentation in an exported component, and run `npm run docs` to rebuild the markdown files
- The change should live-reload in docsify
- Export a new component in `client/components/index.js` (copy order-status for a simple one, change some names)
- Run `npm run docs` to rebuild the markdown files
- Check that the new component's docs are built in `/docs/components`, and that it shows up in the docsify site.

**Other notes**

- If this is merged to master, we can set up github pages to [automatically show this website](https://docsify.js.org/#/deploy?id=github-pages), so no need to host it anywhere special 👍 
- The docs generation script only builds `/docs/components`, we can manually write any other markdown files in the `/docs` dir & docsify will load it. We'll need to tweak how the sidebar is generated, though.
- `npm run docs` will now work to build the markdown files. This is not run on any commit hooks, but it could be? It might drastically slow down commits if we grow the components more…
- This doesn't handle live examples or code samples, I was thinking we might leave an `example.js` or `example.md` in each component dir, and read from that when building the markdown files, and inject the content there above the props listing.
